### DISCUSSION TITLE
API rules for limited number of fields in collection

### DISCUSSION THREAD

**Discussion by twiddlecoding** [pocketbase/pocketbase#3595] [Q&A]
Hello!

Let's say we have two collections, posts and users.
The API rules for posts is that everyone can list or view them.
The posts collection has a relation field that points to the user that created it.

I want anyone to be able to extend a fetch to posts with the users name and avatar fields, but not with anything else from the user collection, the rest should be publically hidden. How can I setup the API rules in the users collection to achieve this?

Thanks a lot!
Sebastian

**Comment by ganigeorgiev** [comment]
`expand` will expand only the relations for which the client request satisfies the View API rule of the related collection (in your case "users").

At the moment we don't support hiding/restricting individual fields from the records response, aka. it is either all or nothing.
Depending on your use case, a workaround could be to move the "private" fields in a separate collection (eg. "profiles") and define its own API rules.

Note that you can request to retrieve only a subset of the record fields using `fields: "*,expand.user.name,expand.user.avatar"`, but this is just client-side parameter that everyone can set to minimize the returned json response size (_another user suggested in the past and I have it somewhere buried in my todo to consider adding a collection option to allow defining an optional admin level fields-rule pairs but this could have negative impact on the performance and for now is left aside_).
